### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by kryvets-vadym**
[DEMO LINK](https://kryvets-vadym.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const visibleMovies = [...moviesFromServer];
+
+  const [movies, setMovies] = useState(visibleMovies);
+
+  const addMovie = (newMovie: Movie) => {
+    setMovies(currentMovies => [...currentMovies, newMovie]);
+  };
```
Comment (denys-danyliuk) [inline src/App.tsx:15]:
```suggestion
  const addMovie = (newMovie: Movie) => {
    setMovies([...currentMovies, newMovie]);
  };
```

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,104 @@
...
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isImgUrlValid = isUrlValid(imgUrl);
+  const isImdbUrlValid = isUrlValid(imdbUrl);
+
+  const submitApprove = title.trim().length > 0
```
Comment (denys-danyliuk) [inline src/components/NewMovie/NewMovie.tsx:28]:
```suggestion
  const isDisabled = !(title.trim()
```

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,104 @@
...
+
+    onAdd(newMovie);
+
+    setCount(prevValue => prevValue + 1);
+
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
```
Comment (denys-danyliuk) [inline src/components/NewMovie/NewMovie.tsx:52]: move this to a function

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]
add an error when field filled with spaces, like title in this example
Знімок екрана 2022-12-06 о 11 19 42

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**